test/src/main/java/org/apache/accumulo/test/functional/ReadLargeKVIT.java

```diff
@@ -0,0 +1,167 @@
+  public void test() throws Exception {
+
+    final int NUM_ROWS = 10;
+    byte[] EMPTY = new byte[0];
+    byte[] VALUE = new byte[11 * 1024 * 1024];
+    Arrays.fill(VALUE, (byte) '1');
+
+    final AtomicReference<Throwable> ERROR = new AtomicReference<>();
+    String table = getUniqueNames(1)[0];
+
+    Thread t = new Thread(() -> {
+      try (AccumuloClient client2 = Accumulo.newClient().from(getClientProps()).build()) {
+        try (Scanner scanner2 = client2.createScanner(table)) {
+          IteratorSetting is = new IteratorSetting(11, SlowMemoryConsumingIterator.class,
+              Map.of("sleepTime", "30000"));
+          scanner2.addScanIterator(is);
+          Iterator<Entry<Key,Value>> iter2 = scanner2.iterator();
+          iter2.next();
+          if (!iter2.hasNext()) {
+            throw new RuntimeException("hasNext == false");
+          }
+          LoggerFactory.getLogger(ReadLargeKVIT.class).info("Reached the end");
+        } catch (Exception e) {
+          LoggerFactory.getLogger(ReadLargeKVIT.class).error("Error in thread", e);
+          e.printStackTrace();
+          ERROR.set(e);
+        }
+      }
+    });
+
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+
+      TableOperations to = client.tableOperations();
+      to.create(table);
+
+      // Insert several K/V pairs with 11MB values
+      BatchWriterConfig bwc = new BatchWriterConfig();
+      bwc.setMaxMemory(24 * 1024 * 1024);
+      try (BatchWriter writer = client.createBatchWriter(table, bwc)) {
+        for (int i = 0; i < NUM_ROWS; i++) {
+          Mutation m = new Mutation("test" + i);
+          m.put(EMPTY, EMPTY, VALUE);
+          writer.addMutation(m);
+        }
+      }
+
+      // Validate NUM_ROWS in table
+      long start = System.currentTimeMillis();
+      try (Scanner scanner = client.createScanner(table)) {
+        assertEquals(10, Iterables.size(scanner));
+      }
+      long firstRunTime = System.currentTimeMillis() - start;
+
+      t.start();
+
+      long start2 = System.currentTimeMillis();
+      try (Scanner scanner3 = client.createScanner(table)) {
+        // SlowMemoryConsumingIterator is set to sleep for 30s
+        // and consume all free memory except for 10MB. In theory,
+        // getting the first K/V pair should take close to 30s.
+        scanner3.setBatchSize(1);
+        scanner3.setReadaheadThreshold(1);
+        scanner3.setBatchTimeout(1, TimeUnit.MINUTES);
+        IteratorSetting is =
+            new IteratorSetting(11, SlowMemoryConsumingWaitingIterator.class, Map.of());
+        scanner3.addScanIterator(is);
+        Iterator<Entry<Key,Value>> iter = scanner3.iterator();
+        assertTrue(iter.hasNext());
+        Entry<Key,Value> e = iter.next();
+        assertEquals(VALUE.length, e.getValue().get().length);
+      }
+      long secondRunTime = System.currentTimeMillis() - start2;
+
+      assertNull("Error in background thread", ERROR.get());
```

Comment:
These parameters will probably need to be swapped. With Junits assert statements, I think the first param should be the Object that you want to make sure is null and the second param should be the error message.